onnxruntime/python/tools/transformers/models/bert/eval_squad.py

```diff
@@ -190,7 +197,11 @@ def main():
         tokenizer.model_max_length = sequence_length
         tokenizer.doc_stride = min(sequence_length // 2, 128)
 
-        ort_model, onnx_path = load_onnx_model(pretrained_model_name, args.onnx, args.use_gpu)
+        use_gpu = False
+        if args.provider is not ("CPUExecutionProvider" or None):
```

Comment:
The logic is not correct here. It evaluate to `args.provider is not "CPUExecutionProvider"`, and string literal cannot use `is not` for equality.
 
```
>>> ("CPUExecutionProvider" or None)
'CPUExecutionProvider'
>>> "CPUExecutionProvider" is not ("CPUExecutionProvider" or None)
<stdin>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
False
>>> "CPUExecutionProvider" is not "CPUExecutionProvider"
<stdin>:1: SyntaxWarning: "is not" with a literal. Did you mean "!="?
False
```

You just write
```py
use_gpu = args.provider in ["CUDAExecutionProvider", "ROCMExecutionProvider"]
```

Reply:
Cleaning these to be similar to this PR #13524  since this was a cherry pick of those changes originally.